src/core/src/VertexAttributeType.cpp

```diff
@@ -0,0 +1,113 @@
+#include "cesium/omniverse/VertexAttributeType.h"
+
+#include <CesiumGltf/Accessor.h>
+
+namespace cesium::omniverse {
+
+namespace {
+
+uint64_t getComponentCount(VertexAttributeType type) {
+    switch (type) {
+        case VertexAttributeType::UINT8:
+        case VertexAttributeType::INT8:
+        case VertexAttributeType::UINT16:
+        case VertexAttributeType::INT16:
+        case VertexAttributeType::FLOAT32:
+            return 1;
+        case VertexAttributeType::VEC2_UINT8:
+        case VertexAttributeType::VEC2_INT8:
+        case VertexAttributeType::VEC2_UINT16:
+        case VertexAttributeType::VEC2_INT16:
+        case VertexAttributeType::VEC2_FLOAT32:
+            return 2;
+        case VertexAttributeType::VEC3_UINT8:
+        case VertexAttributeType::VEC3_INT8:
+        case VertexAttributeType::VEC3_UINT16:
+        case VertexAttributeType::VEC3_INT16:
+        case VertexAttributeType::VEC3_FLOAT32:
+            return 3;
+        case VertexAttributeType::VEC4_UINT8:
+        case VertexAttributeType::VEC4_INT8:
+        case VertexAttributeType::VEC4_UINT16:
+        case VertexAttributeType::VEC4_INT16:
+        case VertexAttributeType::VEC4_FLOAT32:
+            return 4;
+        default:
+            assert(false);
```

Comment:
If this is an error case we should probably also log an error to the console so we can diagnose it if someone runs into it.

Reply:
I'm not sure if it's worth logging an error here since this code can only be reached if `type` not one of the enum values. The only time this could happen is if we reinterpet_cast an arbitrary int to `VertexAttributeType`. It's extremely unlikely that would happen, even unintentionally.

I did move the assert outside the switch/case though because we don't want anything falling through to the `default `case.